openarc/openarc.c

```diff
@@ -3646,13 +3651,32 @@ mlfi_eom(SMFICTX *ctx)
 		/*
  		**  Authentication-Results
 		*/
+		int arcchainlen = arc_chain_custody_str(afc->mctx_arcmsg,
+		                                        arcchainbuf,
+		                                        sizeof(arcchainbuf));
+
+		if (arcchainlen >= ARC_MAXHEADER + 1)
```

Comment:
if (arcchainlen >= sizeof(arcchainbuf))

This way you only ever need to change its size in one place.